woocommerce/woocommerce-android
# Issue/5317 extract logic out of presenter

**Part I** from : #5317

### Description

- Extract logic out of `MyStorePresenter` into use cases that we can then reuse for ViewModel
- Use Kotlin flows as the way to propagate data from repository to UI layer. 
- Prepare everything to replace `MyStorePresenter` for a new `MyStoreViewModel` that will expose a stateFlow to handle UI updates in a simpler a more reactive way. 


Most relevant changes from this PR will be found in: 
 - [GetStats.kt](https://github.com/woocommerce/woocommerce-android/pull/5361/files#diff-ca535cace97256a2ea83335004d62501566c56c274ea2bdd5f2b79b3998db54cR1)
 - [GetTopPerformers.kt](https://github.com/woocommerce/woocommerce-android/pull/5361/files#diff-2a312ac4cd59b1a1584c651933ecd2920bb821192cfb7a7eb89b1a22adc0c9baR1)
 - [MyStorePresenter.kt](https://github.com/woocommerce/woocommerce-android/pull/5361/files#diff-5c4043a57d3d75d8dfdeaa2763626d260de00d57fe9db1e7dea978c310145685L1)



### Testing instructions
The idea is to review that MyStore tab still works as expected after migrating to flows. Some interesting use cases to test:

- Switch between different date granularity and check that UI is updates properly
- Remove connectivity, change between different tabs, then restore connectivity
- Swipe to refresh forces refresh (skeletons are shown again and data is refreshed)
- Check also jetpack CP plugin (No visitors data available): 

<img src="https://user-images.githubusercontent.com/2663464/143903970-78ad2070-8b1a-47fe-afc9-b3d7ee0bb3af.png" width="250">

- Or no jetpack plugin available at all:

<img src="https://user-images.githubusercontent.com/2663464/143903528-2318c1f4-6aa2-44c7-9c0c-70dd9b0ec122.png" width="250">
